test/case/convention/openshift/golang-osd-operator/01_generated_files_checker

```diff
@@ -15,21 +15,51 @@ for version in v0.15.1 v0.16.0 v0.17.0 v0.17.1 ; do
     sed s/__operator_sdk_version__/${version}/g go.mod.tmpl > go.mod
     export GOPATH=$GOPATH:`pwd`
 
+    # Check failure when project is not committed
     make update-boilerplate
-    make check-generate || echo "Project is not committed. Checks failed as expected"
+    make generate-check || echo "Project is not committed. Checks failed as expected"
```

Comment:
making some research, looks like there is a simpler way making the same : 
```
bdematte@bdematte-mac ~ % ls >/dev/nul && echo "0" || echo "1" 
zsh: operation not permitted: /dev/nul
1
bdematte@bdematte-mac ~ % ls >/dev/null && echo "0" || echo "1" 
0
```

Reply:
Yes, this works in terms of capturing success or failure. It would still be nice to be able to check the output for the failure, but we can do that in a separate PR.

Reply:
I am not in favor of comparing commands output. It makes the testing maintenance a nightmare with generally a very limited added value. 
If we want to go this extra step, we should define exit codes per type of failure rather than grepping logs.

Reply:
IMO it's worth the tradeoff. I'd rather be forced to update strings in automated tests periodically than have false positives creep in and lie dormant indefinitely. Those are really hard to track down. I speak from experience.

Unique error codes could work too. Just keep in mind we're limited to 255 of them.

Reply:
I had very poor experience with log parsing, that's why I have a different opinion. 
Number of error code isn't an issue, as your test should validate something specific ie we could have return code 10 for deps failure no matter the target, and if it is a dep you expect to fail, you can run it directly to have a more precise code.

Reply:
Sounds good; let's do it.